packages/btc/src/transaction/build.ts

```diff
@@ -381,11 +388,11 @@ export class TxBuilder {
 
     return {
       inputsTotal: sumOfInputs,
-      inputsRemaining: sumOfInputs - sumOfOutputs,
-      inputsNeeding: sumOfOutputs > sumOfInputs ? sumOfOutputs - sumOfInputs : 0,
       outputsTotal: sumOfOutputs,
+      inputsRemaining: sumOfInputs - sumOfOutputs,
       outputsRemaining: sumOfOutputs - sumOfInputs,
-      outputsNeeding: sumOfInputs > sumOfOutputs ? sumOfInputs - sumOfOutputs : 0,
+      needReturn: sumOfInputs > sumOfOutputs ? sumOfInputs - sumOfOutputs : 0,
+      needCollect: sumOfOutputs > sumOfInputs ? sumOfOutputs - sumOfInputs : 0,
     };
```

Comment:
I prefer to get `needReturn` and `needCollect` based on `inputsRemaining`.

```typescript
const inputsRemaining = sumOfInputs - sumOfOutputs;
const needReturn = inputsRemaining > 0 ? inputsRemainning : 0;
const needCollect = inputsRemaining < 0 ? -inputsRemaining : 0;
```

I think this is clearer, expressing that the values ​​​​of `needReturn` and `needCollect` depend on how many inputs remain.

Reply:
We can even just calculate `sumOfInputs - sumOfOutputs`, `outputsRemaining` being the negative of the result of this calculation.

Reply:
Yes, when I looked back in the code, I also felt that it was a bit hard to understand, and this is also the reason I changed the naming of the variables here.